java/code/src/com/redhat/rhn/frontend/action/user/AdminUserEditAction.java

```diff
@@ -206,6 +211,30 @@ else if (roleSetting == null && existingRoles.contains(role.getLabel())) {
         return errors;
     }
 
+    private void processRBACGroupAssignments(HttpServletRequest request, User target) {
+        var userGroups = target.getAccessGroups();
+        var currentGroupLabels = userGroups.stream()
+                .map(AccessGroup::getLabel)
+                .collect(Collectors.toUnmodifiableSet());
+
+        Iterator<AccessGroup> iterator = userGroups.iterator();
+        while (iterator.hasNext()) {
+            AccessGroup group = iterator.next();
+            String groupSetting = request.getParameter(ROLE_SETTING_PREFIX + group.getLabel());
```

Comment:
I'm not sure if I fully understand this `ROLE_SETTING_PREFIX`, but this is old school code (17 year old) running on struts. If it worked I'm fine with it :)

Reply:
In the meantime I think I found it in the `userdetails.jsp`, where the checkbox is populated with the role value and a prefix `role_` 
I keeped the comment so you can confirm if that is the case :)

Reply:
This comes from Welder's PR, but he's not around to explain it. It works with no problem, so I'd not touch it :D

Reply:
if it works don't touch it :)